website/integrations: Semaphore (#12515)
goauthentik/authentik: website/integrations/services/semaphore/index.mdx

---
title: Integrate with Semaphore UI
sidebar_label: semaphore
---

# Semaphore UI

<span class="badge badge--secondary">Support level: Community</span>

## What is Semaphore UI

> Semaphore UI is a modern web interface for managing popular DevOps tools.
> -- https://semaphoreui.com/
>

> > This guide explains how to set up Semaphore UI to use authentik as the OAuth provider for the login to the WebGUI.

Reviewer: ```suggestion
> This guide explains how to configure Semaphore UI to use authentik as the OAuth provider for logging in to the Web GUI.
```

Author: oh I used it from my previous PR's that I made here :D (take a look @ engomo f.e.).

## Preparation

> The following placeholders will be used:

Reviewer: ```suggestion
The following placeholders are used in this guide:
```

- `semaphore.company` is the FQDN of the engomo install.
- `authentik.company` is the FQDN of the authentik install.
- `ak.cert` is the self-signed certificate that will be used for the service provider.

> ## Semaphore UI configuration

Reviewer: Did you mean to title this "authentik configuration"?

Reviewer: HI @nicedevil007 help me understand this one.. don't we want line 24 to be "authentik configuration", because this section is about creating the app and provider in authentik. Then line 45 is the Semaphore section, no?

> Create an application and an OAuth2/OpenID provider in authentik. Use the following parameters for the OAuth2/OpenID provider:

Reviewer: ```suggestion
[Create](https://docs.goauthentik.io/docs/add-secure-apps/applications/manage_apps#add-new-applications) an application and an OAuth2/OpenID provider in authentik. Use the following parameters for the OAuth2/OpenID provider:
```

> 

Reviewer: Please switch the order of these sections, so that the application part comes first, then the provider, to reflect the order of the Wizard process.

**Provider:**

- Name: `SP-semaphore`
- Client type: `Confidential`
- Redirect URIs/Origins (RegEx): `https://semaphore.company/api/auth/oidc/authentik/redirect/`
- Signing Key: `ak.cert`

> - Scopes: `authentik default OAuth Mapping: OpenID 'email', OpenID 'openid'` and `OpenID 'profile'`

Author: I like to do it this way so no question leaves open. But I can remove it ofc :)

Take note of the Client ID and Client Secret, you'll need to give them to Semaphore UI in Step 3.

Leave the rest as default values. The durations can be changed as needed.

**Application:**

- Name: `Semaphore UI`
- Slug: `semaphore`
- Launch URL: `https://semaphore.company/`

## Semaphore UI configuration

> Login to your Semaphore UI host via SSH. Edit the `config.json` (should be located under `/etc/semaphore`) file with the texteditor of your choice.

Reviewer: ```suggestion
Log in to your Semaphore UI host via SSH. Edit the `config.json` file (should be located under `/etc/semaphore`) file with the text editor of your choice.
```

> For example just use this command:

Reviewer: ```suggestion
For example, with the nano editor you can use this command:
```

```bash
nano /etc/semaphore/config.json
```

> Before the last curly brace, add this part.

Reviewer: ```suggestion
Before the last curly brace, add the following content:
```

```
"oidc_providers": {
        "authentik": {
                "display_name": "SSO-Login",
                "provider_url": "https://authentik.company/application/o/semaphore/",
                "client_id": "<<< Client ID >>>",
                "client_secret": "<<< Client Secret >>>",
                "redirect_url": "https://semaphore.company/api/auth/oidc/authentik/redirect/",
                "username_claim": "username",
                "name_claim": "name",
                "email_claim": "email",
                "scopes": ["openid", "profile", "email"]
        }
}
```

:::note It is mandatory to include 'authentik' in lowercase letters. There should also be another curly brace above these lines. Make sure to add a `,` after it to maintain proper formatting. :::

> More information on this can be found in the semaphore documentation https://docs.semaphoreui.com/administration-guide/openid/authentik/.

Reviewer: ```suggestion
More information on this can be found in the Semaphore documentation https://docs.semaphoreui.com/administration-guide/openid/authentik/.
```

Leave the rest as default.

> ## Test the login

Reviewer: Thank you for adding this section! I love a good verification section. ;-)

Author: I learned that from you with my "old" engomo PR ;)
